infinite-gas.md

```diff
@@ -0,0 +1,108 @@
+Infinite Gas
+Here we use the construct `#gas` to represent positive infinity, while tracking the gas formula through execution.
+This allows (i) computing final gas used, and (ii) never stopping because of out-of-gas.
+Note that the argument to `#gas(_)` is just metadata tracking the current gas usage, and is not meant to be compared to other values.
+such, any `#gas(G)` and `#gas(G')` are the _same_ positive infinite, regardless of the values `G` and `G'`.
+In particular, this means that `#gas(_) <Int #gas(_) => false`, and `#gas(_) <=Int #gas(_) => true`, regardless of the values contained in the `#gas(_)`.
+
+```k
+requires "evm.md"
+
+module INFINITE-GAS
+    imports INFINITE-GAS-JAVA
+    imports INFINITE-GAS-HASKELL
+endmodule
+
+module INFINITE-GAS-HASKELL [kore]
+    imports INFINITE-GAS-COMMON
+
+    rule #gas(_)  <Int _C => false [concrete(_C), simplification]
+    rule #gas(_) <=Int _C => false [concrete(_C), simplification]
+    rule #gas(_) >=Int _C => true  [concrete(_C), simplification]
+    rule _C <=Int #gas(_) => true  [concrete(_C), simplification]
+endmodule
+
+module INFINITE-GAS-JAVA [kast]
+    imports INFINITE-GAS-COMMON
+    imports K-REFLECTION
+
+    rule #gas(_)  <Int C => false requires #isConcrete(C) [simplification]
+    rule #gas(_) <=Int C => false requires #isConcrete(C) [simplification]
+    rule #gas(_) >=Int C => true  requires #isConcrete(C) [simplification]
+    rule C <=Int #gas(_) => true  requires #isConcrete(C) [simplification]
+endmodule
+
+module INFINITE-GAS-COMMON
+    imports EVM
+
+    syntax Int ::= #gas ( Int ) [function, functional, no-evaluators, klabel(infGas), symbol, smtlib(infGas)]
+ // ---------------------------------------------------------------------------------------------------------
```

Comment:
Move the textual documentation at the top of the file (and the note about positive infinity below) to right before this declaration.